src/common/transformations/src/transformations/sdpa_to_paged_attention/state_management_pattern.cpp

```diff
@@ -191,12 +199,14 @@ ov::pass::StateManagementPattern::StateManagementPattern(ParameterVector& kv_par
                 }
                 return shape[first_element - 1];
             } else {
-                return num_q_heads;
+                return default_heads_num;
             }
         };
 
-        auto num_k_heads = extract_num_kv_heads(k_heads_unsqueeze);
-        auto num_v_heads = extract_num_kv_heads(v_heads_unsqueeze);
+        auto num_k_heads =
+            extract_num_kv_heads(k_heads_unsqueeze, sdpa_node->get_input_tensor(1).get_partial_shape()[-3]);
+        auto num_v_heads =
+            extract_num_kv_heads(v_heads_unsqueeze, sdpa_node->get_input_tensor(1).get_partial_shape()[-3]);
```

Comment:
minor: please declare a variable with this value  `sdpa_node->get_input_tensor(1).get_partial_shape()[-3]`

Reply:
A good catch -- this is a bug. They should be different.